packages/passwordless-auth-construct/src/services/cognito_user_service.ts

```diff
@@ -0,0 +1,141 @@
+// ../amplify-backend/packages/passwordless-auth-construct/lib/sign-up/createUserService.js
+
+import {
+  AdminCreateUserCommand,
+  AdminDeleteUserAttributesCommand,
+  AdminUpdateUserAttributesCommand,
+  CognitoIdentityProviderClient,
+  DeliveryMediumType,
+} from '@aws-sdk/client-cognito-identity-provider';
+import { PASSWORDLESS_SIGN_UP_ATTR_NAME } from '../constants.js';
+import {
+  CreateUserParams,
+  DeliveryMedium,
+  MarkVerifiedAndDeletePasswordlessParams,
+  UserService,
+} from '../types.js';
+import { logger } from '../logger.js';
+/**
+ * A service for interacting with Cognito User Poo
+ */
+export class CognitoUserService implements UserService {
+  cognitoClient: CognitoIdentityProviderClient;
+  /**
+   * Creates a new Cognito User constructor
+   * @param cognitoClient - CognitoIdentity client (optional)
+   */
+  constructor(cognitoClient?: CognitoIdentityProviderClient) {
+    if (cognitoClient) {
+      this.cognitoClient = cognitoClient;
+    }
+  }
+
+  /**
+   * Update user and mark attribute as verified
+   * @param params - MarkAsVerifiedAndDeletePasswordlessAttribute parameters
+   * @param params.username - Username
+   * @param params.attributeName - User attribute to be verified
+   * @param params.userPoolId - UserPool ID
+   * @param params.region - UserPool region
+   */
+  async markAsVerifiedAndDeletePasswordlessAttribute(
+    params: MarkVerifiedAndDeletePasswordlessParams
+  ) {
+    const { username, attributeName, userPoolId, region } = params;
+    const attributeVerified = {
+      Name: attributeName,
+      Value: 'true',
+    };
+
+    const client =
+      this.cognitoClient || new CognitoIdentityProviderClient({ region });
```

Comment:
request: We've been asked to avoid default definitions for services inside the service, can you make this client required in this class and hoist the default to the `custom-auth/index.ts` file?

Reply:
The problem is setting the region on the client, i couldnt see other examples that were using the region